Application/Commands/Enquiry/Respond/DeclineEnquiryResponseCommand.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Application.Common.Interfaces;
+using Application.Common.Models.Enquiry.Respond;
+using Microsoft.Extensions.Logging;
+
+namespace Application.Commands.Enquiry.Respond;
+
+public record DeclineEnquiryResponseCommand(
+    string SupportReferenceNumber,
+    string TuitionPartnerSeoUrl) : IRequest<bool>
+{ }
+
+public class DeclineEnquiryResponseCommandHandler : IRequestHandler<DeclineEnquiryResponseCommand, bool>
+{
+    private readonly IUnitOfWork _unitOfWork;
+
+    private readonly ILogger<DeclineEnquiryResponseCommandHandler> _logger;
+
+    public DeclineEnquiryResponseCommandHandler(IUnitOfWork unitOfWork,
+        ILogger<DeclineEnquiryResponseCommandHandler> logger)
+    {
+        _unitOfWork = unitOfWork;
+        _logger = logger;
+    }
+
+    public async Task<bool> Handle(DeclineEnquiryResponseCommand request, CancellationToken cancellationToken)
+    {
+        var result = new ResponseConfirmationModel();
+
+        var validationResult = ValidateRequest(request);
+        if (validationResult != null)
+        {
+            validationResult = $"The {nameof(DeclineEnquiryResponseCommandHandler)} {validationResult}";
+            _logger.LogError(validationResult);
+            throw new ArgumentException(validationResult);
+        }
+
+        var tpEnquiry = await _unitOfWork.TuitionPartnerEnquiryRepository
+            .SingleOrDefaultAsync(x => x.Enquiry.SupportReferenceNumber == request.SupportReferenceNumber &&
+                                       x.TuitionPartner.SeoUrl == request.TuitionPartnerSeoUrl, "Enquiry",
+                true, cancellationToken);
+
+        if (tpEnquiry == null)
+        {
+            var errorMessage = $"Unable to find TuitionPartnerEnquiry with Support Ref ('{request.SupportReferenceNumber}') and Tuition Partner SeoUrl ('{request.TuitionPartnerSeoUrl}')";
+            _logger.LogError(errorMessage);
+            throw new ArgumentException(errorMessage);
+        }
+
+        if (tpEnquiry.TuitionPartnerDeclinedEnquiry)
+        {
+            var errorMessage = $"Previously declined TuitionPartnerEnquiry with Support Ref ('{request.SupportReferenceNumber}') and Tuition Partner SeoUrl ('{request.TuitionPartnerSeoUrl}')";
+            _logger.LogError(errorMessage);
+            throw new ArgumentException(errorMessage);
+        }
+
+        tpEnquiry.TuitionPartnerDeclinedEnquiry = true;
+        tpEnquiry.TuitionPartnerDeclinedEnquiryDate = DateTime.UtcNow;
+
+        await _unitOfWork.Complete();
```

Comment:
I suggest we add a try catch here and catch any exception and log and throw appropriate error.

Reply:
I don't think that we should be catching and logging unless there is a good reason for doing so - e.g. you're having to handle the exception in some way or logging additional data.  Since when it propagates up to the UI it will get logged anyway.

But this did highlight that in a few of the other command handlers we were doing that, without adding any specific logging data or handling the exception.  So I have removed the unnecessary error handling from those, so it's cleaner and the handlers are aligned.  If you look at the commit and let me know if you have issues with it.